src/components/vm/consoles/vncEdit.jsx

```diff
@@ -0,0 +1,127 @@
+import PropTypes from 'prop-types';
+import { Button, Form, Modal, ModalVariant } from "@patternfly/react-core";
+
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { DialogsContext } from 'dialogs.jsx';
+import { VncRow } from './vncBody.jsx';
+import { domainChangeVncSettings, domainGet } from '../../../libvirtApi/domain.js';
+
+const _ = cockpit.gettext;
+
+export class EditVNCModal extends React.Component {
+    static contextType = DialogsContext;
+
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            dialogError: undefined,
+            saveDisabled: false,
+            vmName: props.vmName,
+            vmId: props.vmId,
+            connectionName: props.connectionName,
+            vncAddress: props.consoleDetail.address || "",
+            vncPort: props.consoleDetail.port || "",
+            vncPassword: props.consoleDetail.password || "",
+        };
+
+        this.save = this.save.bind(this);
+        this.onValueChanged = this.onValueChanged.bind(this);
+        this.dialogErrorSet = this.dialogErrorSet.bind(this);
+    }
+
+    onValueChanged(key, value) {
+        const stateDelta = { [key]: value };
+        this.setState(stateDelta);
+    }
+
+    dialogErrorSet(text, detail) {
+        this.setState({ dialogError: text, dialogErrorDetail: detail });
+    }
+
+    save() {
+        const Dialogs = this.context;
+
+        const vncParams = {
+            connectionName: this.state.connectionName,
+            vmName: this.state.vmName,
+            vncAddress: this.state.vncAddress || "",
+            vncPort: this.state.vncPort || "",
+            vncPassword: this.state.vncPassword || "",
+        };
+
+        domainChangeVncSettings(vncParams)
+                .then(() => {
+                    domainGet({ connectionName: this.state.connectionName, id: this.state.vmId });
+                    Dialogs.close();
+                })
+                .catch((exc) => {
+                    this.dialogErrorSet(_("VNC settings could not be saved"), exc.message);
+                });
+    }
+
+    render() {
+        const Dialogs = this.context;
+        const { idPrefix } = this.props;
+
+        const defaultBody = (
+            <Form onSubmit={e => e.preventDefault()} isHorizontal>
+                <VncRow idPrefix={idPrefix}
+                                 dialogValues={this.state}
+                                 onValueChanged={this.onValueChanged} />
+            </Form>
+        );
+        const showWarning = () => {
+        };
+
+        return (
+            <Modal position="top" variant={ModalVariant.medium} id={`${idPrefix}-dialog`} isOpen onClose={Dialogs.close} className='vnc-edit'
+                   title={_("Edit VNC settings")}
+                   footer={
+                       <>
+                           <Button isDisabled={this.state.saveDisabled} id={`${idPrefix}-save`} variant='primary' onClick={this.save}>
```

Comment:
saveDisabled is initialized to `false` and never changed. This was probably meant to be used with some form validation? But all fields are optional, right?

Reply:
In the next "consoles: Improve VNC dialogs" commit there's actually some input validation, but it still doesn't disable the Save button AFAICS.

Reply:
Yes, good point. Fixed, also for "Add VNC". (If only we didn't have to implement all this behavior from scratch every time... :-)